src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -237,21 +237,20 @@ trait AttributeComponent {
       def deleteFromMasterAction(ownerIds: Seq[OWNER_ID]) =
         concatSqlActions(sql"""delete a from #${baseTableRow.tableName} a
                 left join #${baseTableRow.tableName}_TEMP ta
-                on (a.name,a.owner_id,a.list_index)<=>(ta.name,ta.owner_id,ta.list_index)
+                on (a.name,a.owner_id,ifnull(a.list_index,-2))=(ta.name,ta.owner_id,ifnull(ta.list_index,-2))
                 where ta.owner_id is null and a.owner_id in """, ownerIdTail(ownerIds)).as[Int]
       def updateInMasterAction(ownerIds: Seq[OWNER_ID]) =
-        concatSqlActions(sql"""update #${baseTableRow.tableName} a
-                left join #${baseTableRow.tableName}_TEMP ta
-                on (a.name,a.owner_id,a.list_index)<=>(ta.name,ta.owner_id,ta.list_index)
-                set a.value_string=ta.value_string, a.value_number=ta.value_number, a.value_boolean=ta.value_boolean, a.value_entity_ref=ta.value_entity_ref
-                where ta.owner_id is not null and a.owner_id in """, ownerIdTail(ownerIds)).as[Int]
+        sql"""update #${baseTableRow.tableName} a
+                join #${baseTableRow.tableName}_TEMP ta
+                on (a.name,a.owner_id,ifnull(a.list_index,-2))=(ta.name,ta.owner_id,ifnull(ta.list_index,-2))
```

Comment:
I'd suggest that comment about -2 belongs in the code instead of the PR.

To avoid tweaking this each time in two separate locations, you might try:
```scala
val joinExpression = "(a.name,a.owner_id,ifnull(a.list_index,-2))=(ta.name,ta.owner_id,ifnull(ta.list_index,-2))"
// ...
on #${joinExpression}
```
just to see how it looks, then use your judgement as to whether it hurts readability.